kattis2canvas.py

```diff
@@ -254,18 +254,29 @@ def course2canvas(offering, canvas_course, dryrun, force, add_to_module):
         if ag.name == 'kattis':
             kattis_group = ag
             break
-
     if not kattis_group:
-        error(f"no kattis assignment group in {canvas_course}")
-        exit(4)
+        # create assignment group if not present on canvas
+        if dryrun:
+            info(f"would create assignment group {kattis_group}.")
+        else:
+            kattis_group = course.create_assignment_group(name='kattis')
+            info(f"created assignment group {kattis_group}.")
 
     if add_to_module:
         modules = {m.name: m for m in course.get_modules()}
         if add_to_module in modules:
             add_to_module = modules[add_to_module]
         else:
-            error(f'could not find {add_to_module} in {modules.keys()}')
-            exit(4)
+            if dryrun:
+                info(f"would create and publish {add_to_module}.")
+            else:
+                args = {"name": add_to_module}
+                add_to_module = course.create_module(module=args)
+                info(f"created module {add_to_module}.")
+
+                args = {'published': "true"}
```

Comment:
would it work if you add the args on line 273?  and then you wouldn't have to do the edit? (it's fine the way it is. i was just wondering if we could simplify more.)

Reply:
Even I tried it but,
The Create API does not have a **published** parameter. I am not sure why but yeah, so we have to edit the module after creation. It was not possible in one single step.